src/WebJobs.Extensions.NotificationHub/NotificationHubAttribute.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.NotificationHubs;
+using System;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Attribute used to bind a parameter to an Azure NotificationHub
+    /// </summary>
+    /// <remarks>
+    /// The method parameter type can be one of the following:
+    /// <list type="bullet">
+    /// <item><description><see cref="ICollector{T}"/>, where T is either <see cref="TemplateNotification"/> or <see cref="Notification"/>.</description></item>
+    /// <item><description><see cref="IAsyncCollector{T}"/><see cref="TemplateNotification"/> or <see cref="Notification"/>.</description></item>
+    /// <item><description>out T, where T is either <see cref="TemplateNotification"/> or <see cref="Notification"/>.</description></item>
+    /// <item><description>out T[], where T is either <see cref="TemplateNotification"/> or <see cref="Notification"/>.</description></item>
```

Comment:
you now support `out string` and `out IDictionary<string, string>` -- should document them here and how you parse them (i.e. explain that the string must be a JSON object)